OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -52,28 +54,29 @@ void TracerThread::Run(
                                                BIG_RING_BUFFER_SIZE_KB};
         if (uprobe_ring_buffer.IsOpen()) {
           fds_to_ring_buffer.emplace(uprobe_fd, std::move(uprobe_ring_buffer));
-          uprobe_fds_to_function.emplace(uprobe_fd, &function);
+          uprobe_fds_to_function_.emplace(uprobe_fd, &function);
         }
 
         int uretprobe_fd = uretprobes_event_open(
             function.BinaryPath().c_str(), function.FileOffset(), -1, cpu);
-        PerfEventRingBuffer uretprobe_ring_buffer{uretprobe_fd,
-                                                  SMALL_RING_BUFFER_SIZE_KB};
-        if (uretprobe_ring_buffer.IsOpen()) {
-          fds_to_ring_buffer.emplace(uretprobe_fd,
-                                     std::move(uretprobe_ring_buffer));
-          uretprobe_fds_to_function.emplace(uretprobe_fd, &function);
-        }
+        uret_probes_fds.push_back(uretprobe_fd);
+
+        // Redirect uretprobes to uprobe ring buffer to reduce number of ring
+        // buffers and to coalesce closely related events.
+        int ret = ioctl(uretprobe_fd, PERF_EVENT_IOC_SET_OUTPUT, uprobe_fd);
+        if (ret) ERROR("PERF_EVENT_IOC_SET_OUTPUT error: %i.\n", ret);
```

Comment:
We don't have a code style (yet?), but I would write this as follows for, IMO, easier readability:

```suggestion
        if (ret) {
          ERROR("PERF_EVENT_IOC_SET_OUTPUT error: %i.\n", ret);
        }
```

Reply:
Also I think that for clarity this should be 
if (ret != 0)

Reply:
Done.